app/app/src/main/kotlin/com/hedvig/app/feature/referrals/ui/tab/ReferralsViewModel.kt

```diff
@@ -2,72 +2,217 @@ package com.hedvig.app.feature.referrals.ui.tab
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.hedvig.android.apollo.OperationResult
-import com.hedvig.android.core.common.RetryChannel
+import arrow.core.raise.either
+import com.hedvig.android.language.LanguageService
 import com.hedvig.app.feature.referrals.data.ReferralsRepository
+import com.hedvig.app.util.apollo.toMonetaryAmount
 import giraffe.ReferralTermsQuery
 import giraffe.ReferralsQuery
-import kotlinx.coroutines.flow.Flow
+import giraffe.fragment.ReferralFragment
+import java.util.*
+import javax.money.MonetaryAmount
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.WhileSubscribed
-import kotlinx.coroutines.flow.combine
-import kotlinx.coroutines.flow.flow
-import kotlinx.coroutines.flow.onEach
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.flow.update
 import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.launch
 
 class ReferralsViewModel(
   private val referralsRepository: ReferralsRepository,
   private val getReferralTermsUseCase: GetReferralsInformationUseCase,
+  private val languageService: LanguageService,
 ) : ViewModel() {
-  private val retryChannel = RetryChannel()
 
-  private val isLoading = MutableStateFlow(false)
+  private val _uiState = MutableStateFlow(ReferralsUiState())
+  val uiState: StateFlow<ReferralsUiState> = _uiState.stateIn(
+    viewModelScope,
+    SharingStarted.WhileSubscribed(5.seconds),
+    ReferralsUiState(),
+  )
```

Comment:
In this case since `_uiState` already is a `StateFlow`, `stateIn` won't do anything, since that one usually takes a `Flow<>` and turns it into a `StateFlow<>`.
So if we are going with the normal always hot `StateFlow` in this screen we can omit the `stateIn()` and instead replace it with just `_uiState.asStateFlow()`

Reply:
I see! Will probably rewrite this to using molecule after new design is in place :)

Reply:
If you wanna try and do something like that together some time in the future I am 100% up for it. Gonna be a very fun exercise so we both learn how to use it better 🤩

Reply:
Yeah absolutely! 👍